### PR TITLE
Fortran

### PR DESCRIPTION
**Pull request by Xiaoxuan1121**
I still believe the problem arises from the process of choosing lambda: I don't quite follow how gglasso chooses lambda, especially when l = 2, so I comment out that part (line 431-439, line 669-677) when I copied it into our scripts. I still keep the special treatment when l = 1 from sparsegl.f90 (line 524-530, line 758-764) for the sake of not restricting `asparse` equal to other values other than 0. To be honest, I think it makes more sense compared to gglasso.

I also observed that the function getoutput in model.R, which is called from glint-utilies.R always increases the first lambda value of the lambdas vector a little bit and the remaining (99) lambdas unchanged (the discrepancy comes from the raw output alam from sparsegl.f90/log.f90 and the fitobj$lambda).

Except this problem, if you preset a vector of lambdas to the model and let X be a sparse matrix, sparsegl(X, y, loss = "logit") and sparsegl(as.matrix(X), y, loss = "logit") always return the same outputs. (I know I couldn't justify this with only some sample datasets I keep using but rigorous check with packages).

Edit: R doesn't crash anymore after I realized I do try to access unallocated space before.

### CODE REVIEW

**Comment by dajmcdon** [comment]
Merged the fortran files manually.
